presto-spark-base/src/main/java/com/facebook/presto/spark/execution/http/PrestoSparkHttpWorkerClient.java

```diff
@@ -199,8 +220,79 @@ public ListenableFuture<BaseResponse<TaskInfo>> updateTask(
                 createAdaptingJsonResponseHandler(taskInfoCodec));
     }
 
+    public ListenableFuture<BaseResponse<ServerInfo>> getServerInfo()
+    {
+        Request request = setContentTypeHeaders(false, prepareGet())
+                .setUri(serverUri)
+                .build();
+        return httpClient.executeAsync(request, createAdaptingJsonResponseHandler(serverInfoCodec));
+    }
+
+    public SettableFuture<ServerInfo> getServerInfoWithRetry()
```

Comment:
Yeah so is this one. I think this one calls the below private method. client class should only deal with simple http calls, sync or async.